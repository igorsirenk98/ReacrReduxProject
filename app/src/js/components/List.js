import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchProducts } from '../actions'

const mapStateToProps = state => ({
    products: state.products.products,
    error: state.products.error
});

class List extends Component {
    componentDidMount() {
        this.props.dispatch(fetchProducts());
    }

    render() {
        const { products, error } = this.props;

        if (error) {
            return <div>Error! {error.message}</div>;
        }

        return (
            <ul>
                {products.map(item => (
                    <li key={item.productId}>
                        {item.name}
                        <img src={`data:image/jpeg;base64, ${item['productProductPhoto.productPhoto.largePhoto']}`}/>
                    </li>
                ))}
            </ul>
        ); 
    }
}

export default connect(mapStateToProps)(List);