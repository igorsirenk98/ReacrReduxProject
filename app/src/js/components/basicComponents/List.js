import React from 'react';
import { Link } from 'react-router-dom';

import { ProductCard } from './ProductCard';

import CssBaseline from '@material-ui/core/CssBaseline';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';

const listStyles = {
    'display': 'flex',
    'flexWrap': 'wrap',
    'justifyContent': 'space-around'
};

export const List = props => {
    const products = props.products;

    return (
        <ul style={listStyles}>
            {products.map(product => (
                <>
                    <CssBaseline />
                    <Container fixed>
                        <ProductCard key={product.productId} product={product} />
                    </Container>
                </>
            ))}
        </ul>
    )
}
