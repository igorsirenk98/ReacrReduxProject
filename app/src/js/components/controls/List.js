import React from 'react';

import CssBaseline from '@material-ui/core/CssBaseline';
import Container from '@material-ui/core/Container';

import ProductCard from './ProductCard';

const List = (props) => {
    const { products } = props;

    return (
        <>
            <CssBaseline />
            <Container className="container" fixed>
                <ul className="products__list">
                    {products.map((product) => (
                        <ProductCard key={product.productId} product={product} />
                    ))}
                </ul>
            </Container>
        </>
    );
};

export default List;
