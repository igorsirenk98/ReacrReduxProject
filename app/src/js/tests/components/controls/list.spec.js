import React from 'react'
import { shallow } from 'enzyme'
import List from '../../../components/controls/List';

describe('List component', () => {
    const list = shallow(<List products={[{productId: 1}, {productId: 2}]} />);

    it('renders correctly', () => {
        expect(list.debug()).toMatchSnapshot()
    })
})
