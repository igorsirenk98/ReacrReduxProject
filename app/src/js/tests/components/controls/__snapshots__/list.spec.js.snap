// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders correctly 1`] = `
"<Fragment>
  <CssBaseline />
  <WithStyles(ForwardRef(Container)) className=\\"container\\" fixed={true}>
    <ul className=\\"products__list\\">
      <ProductCard product={{...}} />
      <ProductCard product={{...}} />
    </ul>
  </WithStyles(ForwardRef(Container))>
</Fragment>"
`;
